### DISCUSSION TITLE
shelly em controllo assorbimenti

### DISCUSSION THREAD

**Discussion by essegi65** [indomus/forum#334] [HUB: Home Assistant]
Buongiorno a tutti. Ho installato, in sostituzione di un PZEM004, uno shelly em appena dopo il contatore.
Da una settimana controllando lo storico degli assorbimenti vedo che ci sono delle punte che arrivano anche a 6kw, il mio contatore però è di potenza massima 3kw.
Inoltre anche la notte quando in casa rimane acceso solo il frigorifero, il modem e il RPI4 nello storico trovo valori con punte anche di 4kw. Lo shelly em è collegato a HA tramite MQTT e configurato nel file configuration.yaml, ho provato anche a sentire shelly italia ma mi dicono che essendo collegato tramite mqtt potrebbe dipendere dal file di configurazione (che peraltro io ho controllato con quello riportato su indomus ed è identico).
Qualcuno riesce ad aiutarmi per capire dove è il problema?
Grazie

**Comment by KC-inDomus** [comment]
una sola pinza amperometrica? o due

**Reply by essegi65** [reply]
solo una pinza da 50A (collegata correttamente con + e -) ho controllato, più volte per essere sicuro, anche la direzione della freccia sulla pinza.

**Reply by KC-inDomus** [reply]
beh sì, al più avresti valori negativi

**Reply by essegi65** [reply]
Ci potrebbero essere differenze se invece di configurare lo shelly em (con mqtt)  in configuration.yaml usassi l'integrazione nativa di HA?

**Reply by snakuzzo** [reply]
Prova...ma i dati che espone, salvo bug, sono gli stessi

**Reply by essegi65** [reply]
Ho appena fatto la prova escludendo lo shelly da MQTT e inserendo i comandi REST. Da una prima valutazione sembrano molto più corretti adesso che prima (anche l'aggiornamento dei dati di consumo reale avviene molto rapidamente). Ora però con i comandi REST non so come fare il grafico dello storico, perchè mettendo il nome del sensore creato con il platform REST appunto nello storico questo fa vedere solo dei segmenti (anziché il grafico), volevo controllare se anche così lo shelly raggiungeva soglie di 6kw.

**Reply by snakuzzo** [reply]
Ma scusa...non hai detto che volevi usare l'integrazione nativa?

**Reply by essegi65** [reply]
E' vero e sarà la prova successiva ho solo voluto provare anche questo metodo.

**Reply by snakuzzo** [reply]
Resta il fatto che puoi tenere attive assieme tutte le modalità avendo entità diverse per ogni integrazione e vedere quali sono le differenze

**Reply by essegi65** [reply]
grazie difatti penso che farò proprio così. sapete dirmi come fare un grafico da un sensore fatto con i comandi REST?

**Reply by snakuzzo** [reply]
Il grafico viene da se....se definisci il sensore correttamente, definendo un unità di misura...ce l'hai gratis

**Reply by essegi65** [reply]
ma l'unità di misura su platform REST viene rilevata?
Mi sembrava di ricordare che non lo fosse o ricordo male?

**Reply by snakuzzo** [reply]
https://www.home-assistant.io/integrations/sensor.rest/#unit_of_measurement

**Reply by essegi65** [reply]
ok grazie. mi sembrava di ricordare di aver messo una volta l'unità di misura ma non vedevo il grafico, quando sono davanti al pc provo.
ho provato ecco il codice:
```
  - platform: rest
    name: "Consumo Istantaneo"
    scan_interval: 5
    resource: http://192.168.x.xxx/emeter/0
    value_template: "{{ value_json['power'] }}"
    device_class: power
    unit_of_measurement: "W"
```
ma pur avendo inserito nello storico il friendly_name corretto il grafico appare così
![Immagine 2021-05-20 213242](https://user-images.githubusercontent.com/60359328/119038070-fde6a300-b9b2-11eb-9a32-f677879dc607.png)

Dove ho sbagliato?

**Comment by essegi65** [comment]
ho provato ecco il codice:
```
  - platform: rest
    name: "Consumo Istantaneo"
    scan_interval: 5
    resource: http://192.168.x.xxx/emeter/0
    value_template: "{{ value_json['power'] }}"
    device_class: power
    unit_of_measurement: "W"
```
ma pur avendo inserito nello storico il friendly_name corretto il grafico appare così
![Immagine 2021-05-20 213242](https://user-images.githubusercontent.com/60359328/119038433-659cee00-b9b3-11eb-8305-5be1e85d53cd.png)


Dove ho sbagliato?

**Reply by KC-inDomus** [reply]
scusa, questo è il consumo cumulativo, non quello istantaneo

**Reply by essegi65** [reply]
Con il valore "power" francamente viene indicato il consumo istantaneo (difatti cambia di continuo a seconda delle utenze che accendo), mentre con con "total" (o energy) legge il valore del consumo totale.
Comunque grazie lo stesso, come scritto sotto, ho risolto

**Comment by essegi65** [comment]
scusate ho risolto avevo fatto un errore di battitura sul sensore

**Comment by essegi65** [comment]
Buongiorno a tutti, sapreste aiutarmi a capire come mai uno shelly EM mi da 2 letture differenti a seconda di come è ingrato su HA?
Nelle immagini che allego una riporta il grafico derivato dall'integrazione originale SHELLY, l'altro riporta i dati delle letture estrapolate con il comando REST (intervalli di 2secondi) all'indirizzo IP dello shelly.
Grazie a tutti
![integrazioneShelly](https://user-images.githubusercontent.com/60359328/120188884-f62ac800-c216-11eb-9d18-ba0f9c5fd996.jpg)
![restShelly](https://user-images.githubusercontent.com/60359328/120188917-ff1b9980-c216-11eb-88c6-d219470aed87.jpg)

**Reply by herik63** [reply]
quegli spike sono di un motore o simile, a me capita coi frigoriferi.

**Reply by essegi65** [reply]
Mi sembra strano, gli unici motori in casa che potrebbero dare degli spunti sono una autoclave (controllata da un sonoff pow) che non mi da questi spunti e il frigorifero (che però non dovrebbe darli sempre? p.e. anche di notte visto che è acceso 24/24).
Il frigo che assorbe pochi watt, circa 240, come può arrivare ad avere uno spunto di oltre 1,5kw?
Mah il mistero rimane

**Reply by KC-inDomus** [reply]
4 kw di spunto, alla faccia, mi pare troppo

**Reply by directy01** [reply]
Ciao, ho appena istallato uno Shelly EM ed anche i miei grafici mostrano picchi di questo tipo e valore a causa della partenza dei congelatori.

**Reply by essegi65** [reply]
Potresti fare un ulteriore verifica: in orario notturno (se a casa tua è tutto spento tranne i frigoriferi e/o congelatori) vedi sempre dei picchi "importanti" magar intorno ai 2 kw? se si che misura ti da?
Lo shelly dei picchi li deve far vedere specialmente in orari dove la casa è abitata e ci sono elettrodomestici in funzione, lo strano è quando non c'è nessuno in casa o di notte con gli elettrodomestici spenti.

**Comment by duedipicche** [comment]
Posto qui per non aprire un nuovo topic.
Lo Shelly EM od il nuovo Shelly EM Gen3 hanno la lettura della frequenza di rete.
Vorrei sostituire il mio PZEM che la lettura di frequenza ce l'ha, ma non è molto preciso sopra i 2 Kw.
Grazie

**Reply by herik63** [reply]
La frequenza è per forza 50, e ha anche una tolleranza molto molto bassa, se no succede come in Spagna.

**Comment by duedipicche** [comment]
![Immagine 2025-05-11 141955](https://github.com/user-attachments/assets/40eab46c-c3e2-4328-86ae-6d325c97edc3)
Lo so che la frequenza è 50Hz, ma ora riesco a misurarla con PZEM e varia da 49,9 a 50, per la precisione mi riferivo alla misura di potenza, alcune volte sopra i 2Kw improvvisamente va oltre i 6 Kw, ma io ho un contatore da 3 Kw e per questo volevo provare lo Shelly.So che lo Shelly 3EM, non EM Gen3, ha la misura di frequenza, ma è un trifase e sarebbe sprecato per l'impianto di casa mia.
Volevo solo "abbellire" un po' la dashboard con un po' digrafici

**Reply by herik63** [reply]
Che vada fuori così tanto sarebbe fuori da ogni specifica, 
![IMG_0081](https://github.com/user-attachments/assets/52f835e2-2cf9-4ac4-ae1d-b405cffde755)
guarda che il frequenzimetro per valori così bassi solitamente viene moltiplicato x100 o x1000 e poi diviso in visualizzazione, ricordo un generatore audio che aveva il frequenzimetro incorporato e che aveva questo funzionamento, necessario per fare misure su altoparlanti. 
Qui il mio PZEM004 collegato ai frighi, nota gli spike di assorbimento in prossimità dell’accensione del compressore.
![IMG_0081](https://github.com/user-attachments/assets/162c3d5f-e990-4c91-890e-f8527cf871c5)

**Comment by duedipicche** [comment]
Vi ringrazio per le risposte, ma torniamo alla mia domanda iniziale. Lo Shelly EM o lo Shelly EM Gen3 hanno la misura della frequenza? Dalle ricerche in rete non si capisce non trovo la lista dei sensori dei due apparati.Grazie di nuovo

**Reply by KC-inDomus** [reply]
l'EM non mi risulta
